utils/problems.py

```diff
@@ -415,6 +451,34 @@ async def fetch_problems_solved_and_rank(
         submit_stats_global = matched_user["submitStatsGlobal"]
         ac_submission_num = submit_stats_global["acSubmissionNum"]
 
+        languages_problem_count = matched_user["languageProblemCount"]
+
+        tags_problem_count = matched_user["tagProblemCounts"]
+        advanced_tags_count = tags_problem_count["advanced"]
+        intermediate_tags_count = tags_problem_count["intermediate"]
+        fundamental_tags_count = tags_problem_count["fundamental"]
+
+        languages_problem_count_list = []
+
+        for language_problem_count in languages_problem_count:
+            languages_problem_count_list.append(LanguageProblemCount(language = language_problem_count["languageName"], problem_count = language_problem_count["problemsSolved"]))
```

Comment:
This seems like an appropriate place to use list comprehensions, similarly for the advanced_tag_list , etc...